### PR TITLE
Feature/migration logic

### PR DESCRIPTION
**Pull request by joe94**
**Main change introduced in this PR:**
Capturing webhook items migrated by Hecate from SNS to Argus.

**Minor changes:**
- Remove group loggers.
- Log errors when converting items to webhooks
- Don't return partial data: when an item (from among many) fails to convert into a webhook, the whole getAll operation should fail unless we advertise GetAll to be a best effort operation.

### CODE REVIEW

File: service.go [hunk 1]
```diff
@@ -117,66 +187,79 @@ func itemToWebhook(i model.Item) (Webhook, error) {
...
 }
 
-func itemsToWebhooks(items []model.Item) []Webhook {
+func itemsToWebhooks(items []model.Item) ([]Webhook, error) {
 	webhooks := []Webhook{}
 	for _, item := range items {
 		webhook, err := itemToWebhook(item)
 		if err != nil {
-			continue
+			return nil, err
```
Comment (joe94) [inline service.go:260]: if we want to just continue in this case, I would add an error log statement.

File: service.go [hunk 2]
```diff
@@ -64,22 +93,63 @@ func (s *service) Add(owner string, w Webhook) error {
...
 	items, err := s.argus.GetItems(s.config.Bucket, owner)
 	if err != nil {
 		return nil, err
 	}
-	webhooks := []Webhook{}
+
+	for _, item := range items {
+		webhook, err := itemToWebhook(item)
+		if err != nil {
+			return nil, err
```
Comment (joe94) [inline service.go:120]: If we want to continue when seeing a failure, I'd add a log statement here.
FYI: The failure cases are mostly around JSON encoding/decoding.

File: webhook.go [hunk 3]
```diff
@@ -4,26 +4,29 @@ import (
 	"time"
 )
 
+// WebhookConfig is a Webhook substructure with data related to event delivery.
+type WebhookConfig struct {
```
Comment (kristinaspring) [inline webhook.go:8]: Not a huge fan of this name, but not sure what a better name could be.  `DeliveryConfig`? `EventRequestConfig`? But the field is called config and we can't change that so also fine with leaving as is. 🤷

File: service.go [hunk 4]
```diff
@@ -6,12 +6,26 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
-	"strings"
+	"sort"
```
Comment (kristinaspring) [inline service.go:9]: This file needs the copyright comments at the top.  Other files may be missing this too.

File: service.go [hunk 5]
```diff
@@ -35,51 +60,113 @@ type Config struct {
...
+	if err != nil {
+		return err
+	}
+
+	for _, item := range items {
+		webhook, err := itemToWebhook(item)
+		if err != nil {
+			return err
+		}
+		webhookSet[item.ID] = webhook
```
Comment (kristinaspring) [inline service.go:156]: What happens when webhookSet is nil? Why not just return a map? It looks like we're passing an empty one anyways.

File: webhook.go (hunk 3 above)
Comment (joe94) [reply webhook.go:8]: I like `DeliveryConfig` 🚀

File: service.go (hunk 4 above)
Comment (joe94) [reply service.go:9]: Cool, added an issue for it https://github.com/xmidt-org/ancla/issues/30

File: service.go (hunk 5 above)
Comment (joe94) [reply service.go:156]: As we're removing all the migration related code, this will no longer be relevant.

**Comment by joe94** [comment]
Closing this in favor of #31

**Pull request closed without merging**